server/src/routes/series.js

```diff
@@ -0,0 +1,126 @@
+const express = require('express');
+const mongoose = require("mongoose");
+const HttpStatus = require("http-status-codes");
+const Series = require('../db/mongo/models/series');
+const Genres = require('../db/mongo/models/genre');
+
+const { filterSeries, getMostWatchedSeries, aggregateSeries, getCommonSeriesAmongFollowing } = require('../services/series');
+const { getUserWatchlist } = require('../services/watchlist');
+
+const router = express.Router();
+
+const pageLimit = 10;
+
+router.get('/', async (req, res) => {
+    const { pageNumber } = req.query;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        response = await filterSeries(req.body, pageNumber, pageLimit);
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        response = `Failed while fetching genres: ${e}`;
+    }
+
+    res.status(statusCode).send(response);
+});
+
+router.get('/filters', async (req, res) => {
+    let genres;
+    let statusCode = HttpStatus.OK;
+    const statuses = Series.schema.path('status').enumValues;
+    try {
+        genres = await Genres.find();
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        genres = `Failed while fetching genres: ${e}`;
+    }
+
+    res.status(statusCode).send({ genres, statuses });
+});
+
+router.get('/commonAmongFollowing/:email', async (req, res) => {
+    const { pageNumber } = req.query;
+    const { email } = req.params;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        const userWatchList = await getUserWatchlist(email);
```

Comment:
No need to go crazy here. `getUserWatchlist` is quite expensive and the first two lines of it should be more than enough.
You don't need the series name, you need the series _id.
Also, you can call it in `getCommonSeriesAmongFollowing` straight away.

Reply:
i try to call it in getCommonSeriesAmongFollowing  but it doesn't work... i think because circular dependency